onnxscript/rewriter/ort_fusions/fuse_packed_qkv_qga.py

```diff
@@ -71,6 +85,29 @@ def check(
         def no_match(val: ir.Value, dims: Sequence[str]) -> bool:
             return not _fusion_utils._check_shape(self.bindings, val, dims)
 
+        # Check that if x is being split into q, k, v correctly
+        # based on hidden sizes
+        if packed_qkv is None or packed_qkv.shape is None or len(packed_qkv.shape) != 3:
+            return check_result.fail("packed_qkv is not a 3D tensor.", packed_qkv)
+        hidden_size = packed_qkv.shape[2]
+        if not isinstance(hidden_size, int):
+            return check_result.fail("Hidden size is not an integer.", packed_qkv)
+        head_size = hidden_size // (q_num_heads + (2 * kv_num_heads))
+        q_hidden_size = head_size * q_num_heads
+        kv_hidden_size = head_size * kv_num_heads
+        if not (
+            _ir_utils.is_singleton_value(start1, 0)
+            and _ir_utils.is_singleton_value(end1, q_hidden_size)
+            and _ir_utils.is_singleton_value(start2, q_hidden_size)
+            and _ir_utils.is_singleton_value(end2, (q_hidden_size + kv_hidden_size))
+            and _ir_utils.is_singleton_value(start3, (q_hidden_size + kv_hidden_size))
+            and _ir_utils.is_singleton_value(end3, hidden_size)
```

Comment:
In other examples, I have seen the use of MAXINT to slice up to the end. Eg. try what happens if we export `x[from:]`. So, it is better to check if end3 >= hidden_size. (I think I have an example in another rule doing this.)

Reply:
I mean it is ok if end3 >= hidden_size. So, fail if end3 < hidden_size.

Reply:
Updated